### ISSUE TITLE
RFE: Button to Reconnect

### ISSUE THREAD

**Issue by morganfainberg** [jamesrom/jamesrom.github.io#29]
There should be a button to reconnect the websocket forcibly, so you don't lose history but just have a gap of "no data" [e.g. computer went to sleep]

**Comment by jamesrom** [comment]
Agreed. I currently don't handle disconnection very well.

**Comment by lezed1** [comment]
I can take a look at this in the next day or so.

**Comment by lezed1** [comment]
I have a working prototype so far, but there is no marking that distinguishes disconnects. I think I will have a disconnect signaled by a full-height black bar.

**Comment by dougharris** [comment]
Similarly -- I just switched from a wired ethernet connection to wireless while this was open. The bar lost connection and didn't cleanly re-establish its connection. The current bar went to zero and then continued indefinitely into negative numbers.

**Comment by lezed1** [comment]
@dougharris: Yep, that's the current (incorrect) behavior. So far I have working reconnection, but there is no visual marker that the connection was lost. Hopefully I'll fix and push that by tonight.

**Comment by lezed1** [comment]
So I didn't quite get it working by that night, but I didn't forget about this! Here is a picture of what I got working:
![Screenshot](http://i.imgur.com/3s2hgNq.png)
This is an example of a disconnect, then another two disconnects (lose connection, regain, then lose quickly after), and one final one to exhibit the connecting icon.

Disconnects are signaled by a "missing" section of the graph. A 60s click does not erase grid lines, but a disconnect does. I first tried a full sized black bar, but that is too jarring and could be mistaken for a 0s. Thus a gap in the graph looked much nicer.
